vnext/ReactWindowsCore/tracing/fbsystrace.h

```diff
@@ -51,44 +55,48 @@ class FbSystraceSection {
             .count());
   }
 
-  void init(std::string &&v) {
-    args_[index_++] = std::move(v);
-    begin_section();
+  template <typename... RestArg>
+  FbSystraceSection(uint64_t tag, std::string &&profileName, RestArg &&... rest)
+      : tag_(tag), profile_name_(std::move(profileName)) {
+    id_ = s_id_counter++;
+    init(std::forward<RestArg>(rest)...);
   }
 
-  void init(const std::string &v) {
-    args_[index_++] = v; // copy
-    begin_section();
+  ~FbSystraceSection() {
+    end_section();
   }
 
-  void init() {
-    begin_section();
+ private:
+  template <typename T>
+  using ConsructsToString =
+      std::bool_constant<std::is_convertible<T, std::string>::value || std::is_constructible<std::string, T>::value>;
+
+  template <typename Arg, typename std::enable_if_t<ConsructsToString<Arg>::value> * = nullptr>
+  inline void push_arg(Arg &&arg) {
```

Comment:
>) [](start = 32, length = 1)

noexcept?
Generally noexcept functions are 10 times smaller in binary files.

Reply:
I actually disagree about this one.

We're calling throwing code (std::array throws, along with the tracing stuff Anandraj added), which means we the compiler might be generating implicit catch/terminates. These are often times larger than the EH map that would be generated. I know the compiler has some liberty to avoid the implicit catch if it can reason throwing never happens, but no idea how good it is at it.

This is worth it if we're going to be called by noexcept code because we can pay that price in the leaf nodes to allow callers to avoid the overhead, but these functions are exclusively called by Facebook throwing APIs, so we essentially pay the price for a catch for nothing.

We've seen this issue practically in Word. Before a lot of the fixup that happened of shared libraries, building Word's binary with noexcept instead of legacy throw() behavior actually __increased__ binary size due to these implicit catches.

The way I've been thinking to approach this is:
1. If we're sandwiched between throwing code, don't use noexcept. This is both to avoid implicit catch overhead and for correctness/safety.
2. Try to use noexcept everywhere if we have a large object graph that we control and can ensure nothing is mixed internal to the boundary.

This whole thing is also partially mitigated by recent-ish MSVC changes that drastically reduce EH map size on x64.